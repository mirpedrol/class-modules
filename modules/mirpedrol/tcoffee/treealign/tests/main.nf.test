nextflow_process {

    name "Test Process TCOFFEE_TREEALIGN"
    script "../main.nf"
    process "TCOFFEE_TREEALIGN"

    tag "modules"
    tag "modules_mirpedrol"
    tag "tcoffee"
    tag "tcoffee/treealign"
    tag "famsa/guidetree"
    tag "untar"

    test("sarscov2 - fasta - align_with_guide_tree") {

        setup {

            run("FAMSA_GUIDETREE") {
                script "../../../famsa/guidetree/main.nf"
                process {
                    """
                    input[0] = [ [ id:'test' ],
                                    file(params.test_data['sarscov2']['genome']['informative_sites_fas'], checkIfExists: true)
                                ]

                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = [ [ id:'test' ],
                                file(params.test_data['sarscov2']['genome']['informative_sites_fas'], checkIfExists: true)
                            ]
                input[1] = FAMSA_GUIDETREE.out.tree.collect{ meta, tree -> tree }.map{ tree -> [[ id: 'test'], tree]}
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.alignment).match("alignment_guidetree")},
                { assert snapshot(process.out.versions).match("versions_guidetree") }
            )
        }

    }
}