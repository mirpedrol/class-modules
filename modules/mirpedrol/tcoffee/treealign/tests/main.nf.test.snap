{
    "versions_guidetree": {
        "content": [
            [
                "versions.yml:md5,1fe864892229d72f6fe3bca7871491bc"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-11T09:13:29.39446115"
    },
    "alignment_guidetree": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln.gz:md5,93bc8adfcd88f7913718eacc13da8e4a"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-11T09:13:29.320114268"
    }
}